### PR TITLE
Backport PR 4573

### PR DESCRIPTION
**Pull request by kaduk**
OCSP_resp_get0_signer and test

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by mattcaswell** [comment]
Travis red-cross is relevant

**Comment by richsalz** [comment]
its the doc-nits error:
```
doc/crypto/OCSP_resp_find_status.pod:1: get0_signer missing from NAME section
doc/crypto/OCSP_resp_find_status.pod:1: OCSP_resp_get0_signer missing from SYNOPSIS
```

File: doc/crypto/OCSP_resp_find_status.pod [hunk 1]
```diff
@@ -72,7 +76,13 @@ single response B<bs>.
 
 OCSP_resp_get0_certs() returns any certificates included in B<bs>.
 
-OCSP_resp_get0_id() gets the responder id of <bs>. If the responder ID is
+OCSP_resp_get0_signer() attempts to retrive the certificate that directly
```
Comment (FdaSilvaYY) [inline doc/crypto/OCSP_resp_find_status.pod]: s/retrive/retrieve

**Comment by mattcaswell** [comment]
@kaduk - what is the status of this?

**Comment by kaduk** [comment]
@mattcaswell I think I managed to add on all the other relevant follow-up commits that it was waiting on (and rebase).
Note that the cherry-picks are not particularly clean, with (e.g.) 1.1.0 not having the TEST_foo() macros in the test code, etc.

File: crypto/ocsp/ocsp_vfy.c [hunk 2]
```diff
@@ -138,6 +138,15 @@ int OCSP_basic_verify(OCSP_BASICRESP *bs, STACK_OF(X509) *certs,
     goto end;
 }
 
+int OCSP_resp_get0_signer(OCSP_BASICRESP *bs, X509 **signer,
+                     STACK_OF(X509) *extra_certs)
```
Comment (mattcaswell) [inline crypto/ocsp/ocsp_vfy.c]: indentation is off

File: doc/crypto/OCSP_resp_find_status.pod (hunk 1 above)
Comment (mattcaswell) [reply doc/crypto/OCSP_resp_find_status.pod]: This typo is still there

**Comment by kaduk** [comment]
It would seem that I could proceed with matt's approval given that I added a cherry-picked commit that addresses his final concerns, but I will wait a day (okay, 20 hours) to see if there are objections before proceeding.

**Comment by kaduk** [comment]
Thanks again, and sorry this turned into such a marathon.
Pushed, finally, so closing.

**Pull request closed without merging**